dockerfile

```
@@ -0,0 +1,18 @@
+# ---- Build Stage ----
+FROM node:22-alpine AS builder
+WORKDIR /app
+COPY package*.json ./
+RUN npm ci
+COPY . .
+RUN npm run build
+
+# ---- Serve Stage ----
+FROM nginx:1.27-alpine AS server
+RUN apk add --no-cache gettext bash
```

Hadolint (2.12.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)